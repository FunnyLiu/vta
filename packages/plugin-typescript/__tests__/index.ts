import path from "path";
import fs from "fs";
import { run, resolveConfig } from "vta";

jest.setTimeout(100000);

function loadFileContent(file: string): Promise<{ exists: boolean; content?: string }> {
  return new Promise(resolve => {
    fs.exists(file, resolve);
  }).then(exists => {
    if (exists) {
      return new Promise(contentResolve => {
        fs.readFile(file, "utf-8", (err, content) => {
          if (!err) {
            contentResolve({ exists: true, content });
          }
          contentResolve({ exists: false });
        });
      });
    }
    return { exists: false };
  });
}

describe("plugin-typescript", () => {
  it("project-typescript", () => {
    return run({ cwd: path.join(__dirname, "./data/project-typescript"), silent: true })
      .then(({ error: err }) => {
        expect(err).toBe(undefined);
      })
      .then(() => loadFileContent(path.join(__dirname, "./data/project-typescript/dist/delay.js")))
      .then(({ exists, content }) => {
        expect(exists).toBe(true);
        expect(content).toMatchSnapshot("delay");
      })
      .then(() =>
        loadFileContent(path.join(__dirname, "./data/project-typescript/dist/delay.d.ts")),
      )
      .then(({ exists, content }) => {
        expect(exists).toBe(true);
        expect(content).toMatchSnapshot("delay.d");
      })
      .then(() =>
        loadFileContent(path.join(__dirname, "./data/project-typescript/dist/delay.spec.ts")),
      )
      .then(({ exists }) => {
        expect(exists).toBe(false);
      });
  });
  it("project-react", () => {
    return run({ cwd: path.join(__dirname, "./data/project-react"), silent: true })
      .then(({ error: err }) => {
        expect(err).toBe(undefined);
      })
      .then(() => loadFileContent(path.join(__dirname, "./data/project-react/dist/Button.js")))
      .then(({ exists, content }) => {
        expect(exists).toBe(true);
        expect(content).toMatchSnapshot("button");
      })
      .then(() => loadFileContent(path.join(__dirname, "./data/project-react/dist/Button.d.ts")))
      .then(({ exists, content }) => {
        expect(exists).toBe(true);
        expect(content).toMatchSnapshot("button.d");
      });
  });
  it("project-webpack", () => {
    return run({ cwd: path.join(__dirname, "./data/project-webpack"), silent: true })
      .then(({ error: err }) => {
        expect(err).toBe(undefined);
      })
      .then(() => loadFileContent(path.join(__dirname, "./data/project-webpack/dist/delay.js")))
      .then(({ exists }) => {
        expect(exists).toBe(false);
      })
      .then(() => {
        process.env.VTA_JEST_TEST = "true";
        const babelConfig = resolveConfig("babel", path.join(__dirname, "./data/project-webpack"));
        expect(babelConfig.presets[babelConfig.presets.length - 1][0]).toBe(
          "@babel/preset-typescript",
        );
      });
  });
});
