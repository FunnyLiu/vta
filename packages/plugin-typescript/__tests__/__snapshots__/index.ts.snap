// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plugin-typescript project-react: button 1`] = `
"\\"use strict\\";

var _interopRequireDefault = require(\\"@babel/runtime/helpers/interopRequireDefault\\");

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require(\\"react\\"));

var _propTypes = _interopRequireDefault(require(\\"prop-types\\"));

/* eslint-disable */
const Button = ({
  type,
  children
}) => {
  return _react.default.createElement(\\"button\\", {
    type: type
  }, children);
};

Button.propTypes = {
  type: _propTypes.default.oneOf([\\"button\\", \\"submit\\", \\"reset\\"]),
  children: _propTypes.default.node.isRequired
};
var _default = Button;
exports.default = _default;"
`;

exports[`plugin-typescript project-react: button.d 1`] = `
"import React from \\"react\\";
export declare type ButtonType = \\"button\\" | \\"submit\\" | \\"reset\\";
export declare interface ButtonProps {
    type?: ButtonType;
    children: React.ReactNode;
}
declare const Button: React.FunctionComponent<ButtonProps>;
export default Button;
"
`;

exports[`plugin-typescript project-typescript: delay 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = delay;

function delay(ms) {
  return new Promise(resolve => {
    setTimeout(resolve, ms);
  });
}"
`;

exports[`plugin-typescript project-typescript: delay.d 1`] = `
"export default function delay(ms: number): Promise<void>;
"
`;
